dj/sql/parsing/ast.py

```diff
@@ -419,22 +505,36 @@ def __hash__(self) -> int:
 class Column(Named):
     """column used in statements"""
 
-    _table: Optional["Table"] = field(repr=False, default=None)
+    _ref: Union["Table", Namespace, None] = field(repr=False, default=None)
 
     @property
-    def table(self) -> Optional["Table"]:
+    def ref(self) -> Union["Table", Namespace, None]:
         """
-        return the table the column was referenced from
+        return the preceding Namespace or Table if one
```

Comment:
I'm not sure but I think this meant one as in just the table name, so "mytable" instead of "mycatalog.mydb.mytable". Either way this docstring could be clearer.